### PR TITLE
Start of CSV (WIP)

### PR DESCRIPTION
**Pull request by tebeka**

### CODE REVIEW

File: csv/csv.cc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#include "_cgo_export.h"
+#include "csv.h"
+
+// TODO: Unite with the one in carrow.cc
+struct Table {
+  std::shared_ptr<arrow::Table> table;
+};
+
+
+class GoStream: virtual public arrow::io::InputStream {
```
Comment (yonidavidson) [inline csv/csv.cc]: Why virtual inheritance ? isn't it just for multiple inheritance cases ?

**Comment by yonidavidson** [comment]
@tebeka can we discuss the design?

File: csv/csv.cc (hunk 1 above)
Comment (tebeka) [reply csv/csv.cc]: Not sure about `virtual`, that's how I saw it in Python IIRC

**Comment by tebeka** [comment]
> In general - I am not a great fan of inheritance.

It's not inheritance, we're implementing an interface.

> I think that a simpler design would be to pass the csv arrow::io::InputStream pointer and to pass another reader class.

The idea is to provide at the Go level an `io.Reader`. This class implements a callback from C to Go to use this reader.

**Pull request merged**